NPE when combining certain alternative cost KW

Case A:
- _Hunting Velociraptor_
- _Maskwood Nexus_ (so it gets Dinosaur type)
- try cast _Beloved Beggar_ disturbed

Case B:
- Defeat battle _Invasion of Ikoria_ and try cast backside (is already Dinosaur)

```
Game-2 > java.lang.RuntimeException: Error in Keyword Prowl:2 R for card Generous Soul
	at forge.game.keyword.KeywordInstance.createTraits(KeywordInstance.java:128)
	at forge.game.keyword.KeywordInstance.createTraits(KeywordInstance.java:86)
	at forge.game.card.Card.getKeywordForStaticAbility(Card.java:5130)
	at forge.game.card.Card.addChangedCardKeywords(Card.java:5107)
	at forge.game.staticability.StaticAbilityContinuous.applyContinuousAbility(StaticAbilityContinuous.java:772)
	at forge.game.staticability.StaticAbilityContinuous.applyContinuousAbility(StaticAbilityContinuous.java:72)
	at forge.game.staticability.StaticAbility.applyContinuousAbilityBefore(StaticAbility.java:238)
	at forge.game.GameAction.checkStaticAbilities(GameAction.java:1135)
	at forge.game.GameAction.checkStaticAbilities(GameAction.java:1083)
	at forge.game.GameActionUtil.addExtraKeywordCost(GameActionUtil.java:567)
	at forge.player.HumanPlaySpellAbility.playAbility(HumanPlaySpellAbility.java:117)
	at forge.player.HumanPlay.playSpellAbility(HumanPlay.java:106)
	at forge.player.PlayerControllerHuman.playChosenSpellAbility(PlayerControllerHuman.java:1509)
	at forge.game.phase.PhaseHandler.startFirstTurn(PhaseHandler.java:1055)
	at forge.game.GameAction.startGame(GameAction.java:2131)
	at forge.game.Match.startGame(Match.java:90)
	at forge.gamemodes.match.HostedMatch.lambda$1(HostedMatch.java:269)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:842)
Caused by: java.lang.NullPointerException: Cannot invoke "forge.game.spellability.SpellAbility.copyWithDefinedCost(forge.game.cost.Cost)" because the return value of "forge.game.card.CardState.getFirstSpellAbility()" is null
	at forge.game.card.CardFactoryUtil.addSpellAbility(CardFactoryUtil.java:3466)
	at forge.game.keyword.KeywordInstance.createTraits(KeywordInstance.java:117)
	... 19 more
```

the problem is these alternate states don't have a `getFirstSpellAbility()` so the Prowl KW can't use it

@Hanmac
how can we handle this?
write a helper to call instead that could lazy build a SpellPermanent?

--- a/forge-game/src/main/java/forge/game/card/CardFactoryUtil.java
+++ b/forge-game/src/main/java/forge/game/card/CardFactoryUtil.java
@@ -3137,7 +3137,7 @@ public class CardFactoryUtil {
         } else if (keyword.startsWith("Freerunning")) {
             final String[] k = keyword.split(":");
             final Cost freerunningCost = new Cost(k[1], false);
-            final SpellAbility newSA = card.getFirstSpellAbility().copyWithDefinedCost(freerunningCost);
+            final SpellAbility newSA = card.getFirstSpellAbilityWithFallback().copyWithDefinedCost(freerunningCost);
 
             if (host.isInstant() || host.isSorcery()) {
                 newSA.putParam("Secondary", "True");
@@ -3463,7 +3463,7 @@ public class CardFactoryUtil {
         } else if (keyword.startsWith("Prowl")) {
             final String[] k = keyword.split(":");
             final Cost prowlCost = new Cost(k[1], false);
-            final SpellAbility newSA = card.getFirstSpellAbility().copyWithDefinedCost(prowlCost);
+            final SpellAbility newSA = card.getFirstSpellAbilityWithFallback().copyWithDefinedCost(prowlCost);
 
             if (host.isInstant() || host.isSorcery()) {
                 newSA.putParam("Secondary", "True");

--- a/forge-game/src/main/java/forge/game/card/CardFactoryUtil.java
+++ b/forge-game/src/main/java/forge/game/card/CardFactoryUtil.java
@@ -2821,12 +2821,7 @@ public class CardFactoryUtil {
             final String[] k = keyword.split(":");
             final Cost disturbCost = new Cost(k[1], true);
 
-            SpellAbility newSA;
-            if (host.getAlternateState().getType().hasSubtype("Aura")) {
-                newSA = host.getAlternateState().getFirstAbility().copyWithDefinedCost(disturbCost);
-            } else {
-                newSA = new SpellPermanent(host, host.getAlternateState(), disturbCost);
-            }
+            SpellAbility newSA = host.getAlternateState().getFirstSpellAbilityWithFallback().copyWithDefinedCost(disturbCost);
             newSA.setCardState(host.getAlternateState());
 
             StringBuilder sbCost = new StringBuilder("Disturb");

--- a/forge-game/src/main/java/forge/game/card/CardState.java
+++ b/forge-game/src/main/java/forge/game/card/CardState.java
@@ -45,6 +45,7 @@ import forge.game.player.Player;
 import forge.game.replacement.ReplacementEffect;
 import forge.game.spellability.SpellAbility;
 import forge.game.spellability.SpellAbilityPredicates;
+import forge.game.spellability.SpellPermanent;
 import forge.game.staticability.StaticAbility;
 import forge.game.trigger.Trigger;
 import forge.util.ITranslatable;
@@ -375,6 +376,15 @@ public class CardState extends GameObject implements IHasSVars, ITranslatable {
     }
     public final SpellAbility getFirstSpellAbility() {
         return Iterables.getFirst(getNonManaAbilities(), null);
+    }
+
+    public final SpellAbility getFirstSpellAbilityWithFallback() {
+        SpellAbility sa = getFirstSpellAbility();
+        if (sa != null || getTypeWithChanges().isLand()) {
+            return sa;
+        }
+        // this happens if it's transformed backside (e.g. Disturbed)
+        return new SpellPermanent(getCard(), this);
     }
 
     public final boolean hasSpellAbility(final SpellAbility sa) {

